ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -167,14 +168,23 @@ public long getOperationTimeMs() {
   }
 
   @Override
-  public boolean equals(Object thatObject) {
-    if (thatObject instanceof MessageInfo) {
-      MessageInfo that = (MessageInfo) thatObject;
-      return key.equals(that.key) && size == that.size && expirationTimeInMs == that.expirationTimeInMs
-          && isDeleted == that.isDeleted && isTtlUpdated == that.isTtlUpdated && crc == that.crc
-          && accountId == that.accountId && containerId == that.containerId && operationTimeMs == that.operationTimeMs;
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
     }
-    return false;
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    MessageInfo that = (MessageInfo) o;
+    return size == that.size && expirationTimeInMs == that.expirationTimeInMs && isDeleted == that.isDeleted
+        && isTtlUpdated == that.isTtlUpdated && accountId == that.accountId && containerId == that.containerId
+        && operationTimeMs == that.operationTimeMs && Objects.equals(key, that.key) && Objects.equals(crc, that.crc);
```

Comment:
How is this going to be used? Based on the use case, you may not want to compare each field. For example, the `crc` might be optional in the `MessageInfo`. The fact that it is present in only one might not mean that the two `MessageInfo` instances represent different messages.

Reply:
It's to replace a MessageInfo comparison in a closed source transformer implementation test which currently just compares the toString outputs (which outputs a string concatenation of the values of all the fields that are being compared here).  I'm interested in equality of all fields in this scenario.

Reply:
This is a good point - do you want to remove crc out of this or is that necessary?

Reply:
I wanted an exact input == expected result so I'd like to have it in there for now